metricbeat/mb/registry.go

```diff
@@ -144,9 +147,42 @@ func (r *Register) metricSetFactory(module, name string) (MetricSetFactory, Host
 	return info.factory, info.hostParser, nil
 }
 
+//Modules returns the list of module names that are registered
+func (r *Register) Modules() []string {
+	r.RLock()
+	defer r.RUnlock()
+
+	modules := []string{}
+	for module := range r.modules {
+		modules = append(modules, module)
+	}
+
+	return modules
+}
+
+//MetricSets returns the list of metricsets registered for a given module
+func (r *Register) MetricSets(module string) []string {
+	r.RLock()
+	defer r.RUnlock()
+
+	metricsets := []string{}
+
+	sets, ok := r.metricSets[module]
+	if ok {
+		for name := range sets {
```

Comment:
Once it reaches this point, it can allocate the slice with the `metricsets = make([]string, 0, len(sets)`.